### DISCUSSION TITLE
Firebird with DBeaver

### DISCUSSION THREAD

**Discussion by nilooInGitHub** [dbeaver/dbeaver#17129] [DBeaver Development]
Hello Dear Community,

I installed dbeaver and firebird 

I have 2 problems.

1. the stored procedures are not displayed in Navigator?

2. where can I get extensions for debugger for Firebird?

Can someone help me?

**Comment by LonwoLonwo** [comment]
Hello @nilooInGitHub 

What DBeaver version do you use?

Maybe your stored procedures are system procedures? In this case, the Advanced navigation view helps.

![178293162-bcada972-15ae-41ff-b2d7-2f73f7a780f9](https://user-images.githubusercontent.com/45152336/179953705-e60ca44f-4121-4e50-a54c-f8ff840de527.png)

**Reply by nilooInGitHub** [reply]
Hi,
I use Version Version 22.1.2.202207091910  Free Version CE
this Setting didn't help too :(

do you have another Idea?

**Reply by LonwoLonwo** [reply]
You can try to change the driver version in the connection settings. Because we use driver to read procedures list.

![2022-07-20 13_19_51-Connect to a database](https://user-images.githubusercontent.com/45152336/179959110-2eb68618-0ab6-4a62-b849-0e02f8fc0d56.png)

**Reply by nilooInGitHub** [reply]
I have the newest driver installed, this didn't help too. :(

**Reply by LonwoLonwo** [reply]
Ok, and can you please execute this query
`SELECT * FROM RDB$PROCEDURES`
Or check the RDB$PROCEDURES table in the database navigator?
Does this table contain the info about your procedures?

**Reply by nilooInGitHub** [reply]
Dear Anastasiya,

yes, the table contains all infos about the Procedures

**Reply by LonwoLonwo** [reply]
Thanks @nilooInGitHub 

I will transfer this discussion into the ticket.

**Reply by nilooInGitHub** [reply]
Thanks Anastasiya.

I have an other Problem too. :/

why do I get an error message when I use in the statement "with clause"? in the form 

With tablename as ( select .....)  

maybe an extention is missing, which is also responsible for displaying the procedures or some setting.
can you please help me?

**Reply by LonwoLonwo** [reply]
Can you please provide an example of a "with clause" statement?  
And error message

**Reply by nilooInGitHub** [reply]
Hi Anastasiya,
hier an easy Example 

with GRP_MG as (
      
      Select  Bezeichnung , personalnr, abdatum,bisdatum From ZT_MG 
                
    ) 
SELECT * FROM GRP_MG


Error:

SQL-Fehler [335544634] [42000]: Dynamic SQL Error; SQL error code = -104; Token unknown - line 1, column 1; ) [SQLState:42000, ISC error code:335544634]


is this enough or do you need Error details?

**Reply by nilooInGitHub** [reply]
Dear Anastasiya,

Something new about my Problems?
can you please let me know if you know more?

thanks

**Reply by TheMechanist902** [reply]
Hi,
just a guess:
Do you have any empty lines in your query? These are intepreted as command seperators by default. See https://github.com/dbeaver/dbeaver/discussions/17535#discussioncomment-3533433.
I had a similar situation with the WITH clause and disabling that setting solved it.
Maybe it applies to your enviroment as well.
